.github/workflows/self-test-and-release.yml

```diff
@@ -1,10 +1,7 @@
 # Created using https://github.com/aws-samples/aws-codeguru-reviewer-cicd-cdk-sample
 name: Self-test and release
 
-on:
-  push:
-    branches:
-      - main
+on: [push, pull_request, workflow_dispatch]
```

Comment:
Last time I had both of `push` and `pull_request` enabled (unrestricted, like here, for both events in all branches), I quickly realized that doing so triggered a LOT of analyses. Every `push` to a feature branch with an open PR *also* counts as a `pull_request` event, so you end up running 2x Guru for each push to any PR.

Have you considered unrestricted `pull_request`, but `push` only for `main`?